src/factories/siti-agri/agreement.factory.js

```diff
@@ -0,0 +1,91 @@
+import { fakerEN_GB as faker } from '@faker-js/faker'
+import { createLogger } from '../../common/helpers/logging/logger.js'
+import { fakeIds, nullOrFake, transformDate } from '../common.js'
+import { orgIdLookup, sbiToOrgId } from '../id-lookups.js'
+const logger = createLogger()
+
+const agreementsStore = {}
+
+const createPaymentSchedule = (attributes = {}) => {
+  return {
+    option_code: faker.string.alpha({ length: 3 }).toUpperCase(),
+    option_description: faker.lorem.sentence({ min: 3, max: 8 }),
+    commitment_group_start_date: transformDate(faker.date.past({ years: 10 })),
+    commitment_group_end_date: transformDate(faker.date.future({ years: 5 })),
+    year: faker.date.anytime().getFullYear(),
+    sheet_name: faker.string.alpha({ length: 6 }).toUpperCase(),
+    parcel_name: faker.string.numeric({ length: 4 }),
+    action_area: faker.number.int({ min: 1000, max: 50000 }),
+    action_mtl: nullOrFake(() => faker.number.int({ min: 1, max: 100 })),
+    action_units: nullOrFake(() => faker.number.int({ min: 1, max: 100 })),
+    parcel_total_area: faker.number.int({ min: 1000, max: 50000 }),
+    payment_schedule_start_date: transformDate(faker.date.past({ years: 10 })), // replicate kits date format
+    payment_schedule_end_date: transformDate(faker.date.future({ years: 5 })), // replicate kits date format
+    ...attributes
+  }
+}
+
+const createAgreementMock = (sbi, attributes = {}) => ({
+  sbi: `${sbi}`,
+  contract_id: faker.string.numeric({ length: 6 }),
+  agreement_name: faker.helpers.arrayElement([
+    'CS AGREEMENT',
+    'HLS AGREEMENT',
+    'ELS AGREEMENT',
+    'BPS AGREEMENT'
+  ]),
+  status: faker.helpers.arrayElement(['SIGNED', 'ACTIVE', 'EXPIRED', 'WITHDRAWN']),
+  contract_type: faker.helpers.arrayElement([
+    'Countryside Stewardship (MT)',
+    'Higher Level Stewardship',
+    'Entry Level Stewardship',
+    'Basic Payment Scheme'
+  ]),
+  scheme_year: faker.date.anytime().getFullYear(),
+  start_date: transformDate(faker.date.past({ years: 10 })),
+  end_date: transformDate(faker.date.future({ years: 5 })),
+
+  payment_schedules: [],
+  ...attributes
+})
+
+export const retrieveOrganisationAgreements = (sbi) => {
+  if (agreementsStore[sbi]) {
+    return agreementsStore[sbi]
+  }
+  if (!sbiToOrgId[sbi]) {
+    // KITS return error if sbi not found
+    throw Boom.forbidden('Request forbidden by administrative rules.')
+  }
+
+  faker.seed(sbi)
+
+  const fakeAgreements = () =>
+    fakeIds(faker.number.int({ min: 1, max: 3 })).map((id) => {
+      const paymentSchedules = nullOrFake(
+        () => fakeIds(faker.number.int({ min: 1, max: 3 })),
+        0.3333
+      )
+      return {
+        contract_id: `${id}`,
+        payment_schedules: paymentSchedules
+      }
+    })
+
+  const agreements =
+    orgIdLookup[sbiToOrgId[sbi]].agreements || nullOrFake(fakeAgreements, 0.3333) || []
```

Comment:
There's no point using `nullOrFake` in these kinds of contexts, you can just have the array generator in `fakeAgreements` create an array `0+` length. Same for creating payment schedules too.